examples/profile.html

```diff
@@ -0,0 +1,307 @@
+                setHidden(authenticatingEl, true);
+
+                let profileEl = document.querySelector('.user-profile');
+                renderUserProfile(profileEl, profileData);
+                setHidden(profileEl, false);
+            })
+            .catch(function(err) {
+                setHidden(authenticatingEl, true);
+
+                setErrorState(!authenticated ? 'Failed to Authenticate with PureCloud' :
+                    'Failed to fetch/display your profile');
+            });
+
+            // ----- Helper Functions
+
+            /**
+             * Sets the base mode of the app to error and show the provided message
+             */
+            function setErrorState(errorMsg) {
+                let failureEl = document.querySelector('.failure');
+                failureEl.textContent = errorMsg;
+                setHidden(failureEl, false);
+            }
+
+            let profileLinkListener = null;
+            /**
+             * Updates the provided User Profile element state
+             * with the provided profileData
+             */
+            function renderUserProfile(profileRootEl, profileData) {
+                let linkEl = profileRootEl.querySelector('.profile-link');
+
+                if (profileLinkListener) {
+                    linkEl.removeEventListener('click', profileLinkListener);
+                    profileLinkListener = null;
+                }
+
+                if (profileData.id) {
+                    profileLinkListener = function (evt) {
+                        evt.preventDefault();
+                        clientApp.users.showProfile(profileData.id);
+                    }
+
+                    linkEl.addEventListener('click', profileLinkListener);
+                } else {
+                    console.info('No user ID available to route to user profile');
+                }
+
+                let bestImage = {
+                    size: 0,
+                    uri: null
+                };
+                if (profileData.images) {
+                    profileData.images.forEach(function(currImg) {
+                        if (!currImg.resolution || !currImg.imageUri) {
+                            return;
+                        }
+
+                        let currRes = parseInt(currImg.resolution.substr(1), 10);
+                        if (currRes && !isNaN(currRes) && currRes > bestImage.size && currRes <= 400) {
+                            bestImage.size = currRes;
+                            bestImage.uri = currImg.imageUri;
+                        }
+                    });
+                }
+
+                let placeholderPicEl = profileRootEl.querySelector('.placeholder-pic');
+                let picEl = profileRootEl.querySelector('.profile-pic');
+                if (bestImage.size) {
+                    setHidden(placeholderPicEl, true);
+
+                    picEl.setAttribute('src', bestImage.uri);
+                    if (profileData.name) {
```

Comment:
Need an else here since the idea is to have this functionally set the provided element. ... even though it's not a component and wont be reused.